components/registration-framework/org.wso2.carbon.identity.user.registration.engine/src/main/java/org/wso2/carbon/identity/user/registration/engine/validation/InputValidationListener.java

```diff
@@ -47,27 +52,51 @@ public boolean isEnabled() {
     }
 
     @Override
-    public boolean doPostInitiate(RegistrationStep step, RegistrationContext registrationContext)
+    public boolean doPreExecute(RegistrationContext registrationContext)
             throws RegistrationEngineException {
 
-        InputValidationService.getInstance().handleStepInputs(step.getData(), registrationContext);
+        if (MapUtils.isNotEmpty(registrationContext.getUserInputData()) &&
+                MapUtils.isEmpty(registrationContext.getCurrentStepInputs())) {
+            RegistrationGraphConfig registrationGraphConfig = registrationContext.getRegGraph();
+            NodeConfig currentNode = registrationGraphConfig.getNodeConfigs().get(registrationGraphConfig.getFirstNodeId());
```

Comment:
Can we move setting the currentNode to UserRegistrationFlowService. `setCurrentNode`